share/wake/lib/system/job.wake

```diff
@@ -979,10 +979,11 @@ export def makeJSONRunner (plan: JSONRunnerPlan): Runner =
 
             def specFilePath = "{build}/spec-{prefix}.json"
 
-            require Pass inFile =
+            require Pair (Pass inFile) _ =
                 write specFilePath (prettyJSON json)
                 | rmap getPathName
-            else Pair (Fail (makeError "Failed to 'write {specFilePath}'.")) ""
+                | addErrorContext "Failed to 'write {specFilePath}: '"
+                | (Pair _ "")
```

Comment:
What's the `Pair` for?  It seems to be unconditionally added with no meaningful data.

Reply:
The return type of this function is `Pair (Result ...) String` because its a runner pre function. You can see that previously the same Pair was being constructed in the `else` branch. I just made the error message nicer while working around the return type being a Pair

Reply:
Ah, now that I'm seeing the mixed lines, it looks like something to fit the else case.